### DISCUSSION TITLE
Very strange behavior in Flutter_Web

### DISCUSSION THREAD

**Discussion by coogle** [simolus3/drift#3348] [Q&A] [answered]
Hello,

I am experiencing some very strange behavior in flutter web. Unfortunately I haven't been able to create a small reproduction case but basically this is what I am experiencing:

Background: I have a SQLite database that is downloaded and stored in IndexedDb for read-only queries. 

The application is a mapping application and the SQLite database contains points for the map with lat/lon coordinates. When the map renders the **first time** I see the queries are executed as expected, I can also see that the parameters being passed into those prepared statements are also correct. However, I get zero results which is wrong.

Then, I move the map, which triggers the SQLite query to run again.

This time, I see the exact same query come through, with nearly identical results (e.g. I zoom in one click, zoom out one click so I'm back to the starting position)... this time I get results back!

I have been chasing this down everywhere else in my application and at this point am 90% sure this isn't a bug in my code. For some totally unexplained reason, it seems like when the app first loads the widget (and the SQLite database, which is being lazy-loaded via the `get_it` package as `registerLazySingleton()`), the queries somehow silently fail... but then something happens and subsequent queries (even identical ones) return results!

Anyone have any idea why this is happening? I'm stumped.

**Comment by simolus3** [comment]
This sounds odd to me as well - are you loading the initial data with the database (e.g. as an asset you're importing) or are you first opening the database and then inserting the default data?

We shouldn't silently fail queries, they would either respond with an error or the complete result.

**Reply by coogle** [reply, marked as answer]
After much hair-pulling, turns out it wasn't drift after all -- just a very sneaky bug that made drift look like the problem. Thanks!
